src/Dapr.Common/Exceptions/ExtendedErrorDetailFactory.cs

```diff
@@ -0,0 +1,117 @@
+﻿using Google.Protobuf;
+using Google.Protobuf.WellKnownTypes;
+using Google.Rpc;
+
+namespace Dapr.Common.Exceptions
+{
+    /// <summary>
+    /// <see cref="DaprExtendedErrorDetail"/> factory.
+    /// </summary>
+    internal static class ExtendedErrorDetailFactory
+    {
+        private const string DaprErrorTypeUrl = DaprExtendedErrorConstants.ErrorDetailTypeUrl;
+
+        private static Dictionary<string, Func<ByteString, DaprExtendedErrorDetail>> extendedErrorTypeMapping =
+            new()
+            {
+                { $"{DaprErrorTypeUrl}Google.rpc.ErrorInfo", ToDaprErrorInfoDetail },
```

Comment:
This is probably the only bit im actually iffy on; probably just reads better if its a switch assignment in the create

Reply:
As it's an internal class and not publicly out there, I think it's sufficient to swap to a switch statement + a comment indicating it's intended to be a lookup table. I don't envision this code is going to need a lot of changes once merged given the maturity of gRPC.